### PR TITLE
Include all docs in generated HTML

### LINKED ISSUE #535: Documentation does not contain font section
Font related API are not included in the documentation site, but apparently inside the actual header. Maybe there is an option not turned on?

Also, it would be nice if there is a more complete example in the documentation itself.

### PR DESCRIPTION
**Pull request by rongcuid**
Fixes #535

### CODE REVIEW

File: doc/index.html [hunk 1]
```diff
@@ -2260,6 +2270,302 @@
...
+ approximations for value range 0...2*PI. The reason why I decided to
+ approximate exactly that range is that nuklear only needs sine and
+ cosine to generate circles which only requires that exact range.
+ In addition I used Remez instead of Taylor for additional precision:
+ www.lolengine.net/blog/2011/12/21/better-function-approximations.
+ The tool I used to generate constants for both sine and cosine
+ (it can actually approximate a lot more functions) can be
+ found here: www.lolengine.net/wiki/oss/lolremez
         -XXX.XXX-  X...X  -         X...X         -X....X           -           X....X"
 X...XXXXXXXXXXXXX...X -           "
```
Comment (dumblob) [inline doc/index.html:2570]: When at it, could you take a look at this too? I totally forgot about this during those years :cry:. Thanks!

**Comment by dumblob** [comment]
We should remove `index.html` from the root folder and leave only `doc/index.html`.

**Comment by dumblob** [comment]
We should also decide what to do with `doc/nuklear.html`. Suggestions? Probably remove it altogether, right?

**Comment by dumblob** [comment]
Btw. we will need multiple reviewers as this touches `nuklear.h`. Could others please chime in? Thanks!

**Comment by rongcuid** [comment]
Any updates on this?

File: doc/index.html (hunk 1 above)
Comment (rongcuid) [reply doc/index.html:2570]: I don't get what you mean here. This file is generated

**Comment by revolucas** [comment]
> Btw. we will need multiple reviewers as this touches `nuklear.h`. Could others please chime in? Thanks!

Looks okay. The changes to *.h are simply to the comment sections, most likely to make the html generation work.

**Comment by riri** [comment]
LGTM, except the empty index.html created at the root of the project folder, why is so?

**Comment by riri** [comment]
I'm tempted to merge this PR, as all seems ok.

I just have never did it on this project. Is there any preference on how a PR is merged on nuklear?
@dumblob I look at you ^^

**Comment by riri** [comment]
I finally chose to merge this as it has no influence on code base and having this completed doc is a plus!

Thanks again for the contribution @rongcuid

**Comment by dumblob** [comment]
Thanks @riri and everybody!

**Pull request merged**